vassal-app/src/main/java/VASSAL/build/module/PlayerRoster.java

```diff
@@ -483,32 +466,69 @@ protected String promptForSide() {
     }
 
     availableSides.removeAll(alreadyTaken);
+
+    // If a "real" player side is available, we want to offer "the next one" as the default, rather than observer.
+    // Thus hotseat players can easily cycle through the player positions as they will appear successively as the default.
+    // Common names for Solitaire players (Solitaire, Solo, Referee) do not count as "real" player sides, and will be skipped.
+    // If we have no "next" side available to offer, we stay with the observer side as our default offering.
+    boolean found = false;       // Haven't found our own side in the list.
+    String mySide = getMySide(); // Get our own side, so we can find the "next" one
+    String nextChoice = translatedObserver; // For now default to observer
+    if (mySide != null) {                   // If I'm presently an observer, then my next side is just going to be the first real one we find later
+      for (String s : sides) {
+        if (mySide.equals(s)) {             // We're looking for "our" side
+          found = true;                     // We found it, so next "real" one is the one we want
+          continue;            
+        }
+        if (alreadyTaken.contains(s)) {     // Don't want anything that's already taken
+          continue;
+        }
+        if (Resources.getString("PlayerRoster.solitaire").equals(s) ||   // Don't want a solitaire side
+            Resources.getString("PlayerRoster.solo").equals(s) ||
+            Resources.getString("PlayerRoster.referee").equals(s)) {
+          continue;
+        }
+        if (!found) {                       // Don't actually want ANYTHING until we've found "our" current side in the list
+          continue;
+        }
+        nextChoice = s;                     // Aha! We've found a next side after ours, AND it's not taken. THAT's the default we want. 
+        break;
+      }      
+    }
+    
+    if (translatedObserver.equals(nextChoice)) {
+      // If we get in here, that either means "our" side was the last available one we found in the list,
+      // or that there are no other "real" sides left. In either case we now want the first "real" side in the
+      // that's not taken. And if we don't find one we'll just stick with observer as the default.
+      for (String s : availableSides) {
+        if (Resources.getString("PlayerRoster.solitaire").equals(s) ||
+            Resources.getString("PlayerRoster.solo").equals(s) ||
+            Resources.getString("PlayerRoster.referee").equals(s)) {
+          continue;
+        }
+        nextChoice = s;
+        break;
+      }                  
+    }
+
```

Comment:
I think you could to this whole thing more simply like this:
```
  found = false;
  int myidx = sides.indexOf(mySide);
  if (myidx != -1) {
    // we have a side

    int i = (myidx + 1) % sides.size();
    for ( ; i != myidx && !found; i = (i+1) % sides.size()) {
      String s = sides[i];

      if (!alreadyTaken.contains(s) && 
          !Resources.getString("PlayerRoster.solitaire").equals(s) &&
          !Resources.getString("PlayerRoster.solo").equals(s) &&
          !Resources.getString("PlayerRoster.referee").equals(s)) {
        found = true;
      }
    }
  }

  if (found) {
    nextChoice = sides[i];
  }
```

Circling back to the beginning of the list gets folded in as a non-exceptional case this way, and finding the index of the side we already have gets moved outside the loop.

Reply:
We want a "break" in there after the "found = true", right?

Reply:
Oh I see the !found hiding in the for controls

Reply:
You could `break` instead. It's 6 of one, a half dozen of the other for that.

Reply:
Try this out, of course---I typed it up but didn't actually compile or try it.

Reply:
Anyhow, make this less convoluted and I'm ready to merge it.

Reply:
k, yeah turns out we need to break because otherwise "i" gets bollixed on to the next spot in the for-loop logic. Couple other minor complexities I'm ironing out, but it's gonna be a real compact-ass thing compared to before. Will have something up shortly.

Reply:
Yes, breaking is the right thing, and take the `!found` test out of the loop condition. I did say I didn't test this. :)

Reply:
Okay all done & checked in. Runs correctly.